test/controllers/CyaMaintainProfileControllerSpec.scala

```diff
@@ -0,0 +1,287 @@
+          }
+        }
+
+        "must redirect to Journey Recovery if no answers are found" in {
+
+          val application = applicationBuilder(userAnswers = Some(emptyUserAnswers)).build()
+
+          running(application) {
+
+            val request = FakeRequest(GET, routes.CyaMaintainProfileController.onPageLoadNirms.url)
+
+            val result = route(application, request).value
+
+            status(result) mustEqual SEE_OTHER
+            redirectLocation(result).value mustEqual
+              routes.JourneyRecoveryController.onPageLoad(Some(RedirectUrl(journeyRecoveryContinueUrl))).url
+          }
+        }
+
+        "must redirect to Journey Recovery if no existing data is found" in {
+
+          val application = applicationBuilder(userAnswers = None).build()
+
+          running(application) {
+            val request = FakeRequest(GET, routes.CyaMaintainProfileController.onPageLoadNirms.url)
+
+            val result = route(application, request).value
+
+            status(result) mustEqual SEE_OTHER
+            redirectLocation(result).value mustEqual routes.JourneyRecoveryController.onPageLoad().url
+          }
+        }
+      }
+
+      "for a POST" - {
+
+        "when user answers can remove Nirms and update user profile" - {
+
+          "must update the profile and redirect to the Profile Page" - {
+
+            val userAnswers = emptyUserAnswers
+              .set(RemoveNirmsPage, true)
+              .success
+              .value
+              .set(HasNirmsUpdatePage, false)
+              .success
+              .value
+
+            val traderProfile        = TraderProfile(testEori, "1", Some("2"), Some("3"))
+            val updatedTraderProfile = TraderProfile(testEori, "1", None, Some("3"))
+
+            val mockTraderProfileConnector = mock[TraderProfileConnector]
+            val mockAuditService           = mock[AuditService]
+
+            when(mockTraderProfileConnector.getTraderProfile(any())(any())) thenReturn Future.successful(traderProfile)
+            when(mockTraderProfileConnector.submitTraderProfile(any(), any())(any()))
+              .thenReturn(Future.successful(Done))
+            when(mockAuditService.auditMaintainProfile(any(), any(), any())(any))
+              .thenReturn(Future.successful(Done))
+
+            val application = applicationBuilder(userAnswers = Some(userAnswers))
+              .overrides(
+                bind[TraderProfileConnector].toInstance(mockTraderProfileConnector),
+                bind[AuditService].toInstance(mockAuditService)
+              )
+              .build()
+
+            running(application) {
+
+              val request = FakeRequest(POST, routes.CyaMaintainProfileController.onSubmitNirms.url)
+
+              val result = route(application, request).value
+
+              status(result) mustEqual SEE_OTHER
+              redirectLocation(result).value mustEqual routes.ProfileController.onPageLoad().url
```

Comment:
Would probably be nicer to use the Fake navigator here rather than let it go off and do all the actual navigator stuff

Reply:
done 🙂